### DISCUSSION TITLE
Allow specifying ellipsis position for text overrun

### DISCUSSION THREAD

**Discussion by timothyqiu** [godotengine/godot-proposals#9598] [GUI]
When strings like paths are too long, usually the ending of the path is more important than the beginning / middle.

![image](https://github.com/godotengine/godot-proposals/assets/372476/49f80058-7a25-4348-8f04-9180cc16badd)

I suggest allow specifying ellipsis position: begin / middle / end (default, current behavior).

**Comment by Calinou** [comment]
I think core text overrun behavior should remain scoped to basic text (Label's text handling is already pretty complex as it is). It may be better to write a custom solution for file paths specifically.
